src/Foundatio.Extensions.Hosting/Jobs/DynamicJob.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Foundatio.Jobs;
+using Foundatio.Utility;
+
+namespace Foundatio.Extensions.Hosting.Jobs;
+
+internal class DynamicJob : IJob
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly Func<IServiceProvider, CancellationToken, Task> _action;
+
+    public DynamicJob(IServiceProvider serviceProvider, Func<IServiceProvider, CancellationToken, Task> action)
+    {
+        _serviceProvider = serviceProvider;
+        _action = action;
+    }
+
+    public async Task<JobResult> RunAsync(CancellationToken cancellationToken = default)
+    {
+        await _action(_serviceProvider, cancellationToken).AnyContext();
+
```

Comment:
Should this be error handled with logging so we can return cancelled or failure with an error?